### ISSUE TITLE
Code breaking when training TASK16 (Foil).

### ISSUE THREAD

**Issue by iacercalixto** [facebookresearch/vilbert-multi-task#67]
Hi,

I am trying to use this code-base to fine-tune any of the available pretrained models on the FOIL dataset, but I am having some problems. Is the code supposed to work when running the example from the README (script `train_tasks.py`) with `TASK16`?

To reproduce the error, you can run:

    python3 train_tasks.py --bert_model bert-base-uncased --from_pretrained pretrained_model.bin --config_file config/bert_base_6layer_6conect.json --tasks 16 --lr_scheduler 'warmup_linear' --train_iter_gap 4 --task_specific_tokens --save_name finetune_from_gcc

An error happens is in line 363 in `vilbert/task_utils.py`, namely the `vil_binary_prediction` and the `target` tensors' sizes do not match. After much digging, I can make it work when training from scratch by changing a few different files, but I would like to know this head (`vil_binary_prediction`) is supposed to work out-of-the-box.

Thanks!

**Comment by vedanuj** [comment]
I don't think we have tested Task16 support tested properly. Please feel free to open a PR with your changes.

**Comment by zongshenmu** [comment]
How do you resolve the problem? I find the same situation in the NLVR task and doubt the problem in the `vilbert.py`.
